### PR TITLE
docs: Update frontend setup documentation

### PR DESCRIPTION
**Pull request by johnnynotsolucky**
This PR makes some updates to the frontend docs, mostly around initial setup.

Misc changes:
- Added .editorconfig with bare-minimum config;
- Added .nvmrc to pin the node version to v15 - based the version off the version used in the GH workflows.

### CODE REVIEW

File: documentation/src/infrastructure/setup/requirements/frontend.md [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
+    - Or with [nvm](https://github.com/nvm-sh/nvm/blob/master/README.md#installing-and-updating)
+        - `nvm install v15 && nvm alias default v15`
+- rustc nightly _after_ [edition 2021 was stabilized](https://github.com/rust-lang/rust/pull/88100#event-5229813170)
+    - [Install](https://www.rust-lang.org/tools/install)
+    - Update if necessary: `rustup default nightly && rustup update`
+- [cargo-make](https://github.com/sagiegurari/cargo-make)
+    - `cargo install cargo-make`
+- [cargo-watch](https://github.com/watchexec/cargo-watch)
+    - `cargo install cargo-watch`
+- [watchexec-cli](https://github.com/watchexec/watchexec)
```
Comment (johnnynotsolucky) [inline documentation/src/infrastructure/setup/requirements/frontend.md:16]: Doesn't really seem like watchexec is still used by anything on the frontend.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,7 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (MendyBerger) [inline .editorconfig:7]: I think this file should rather go into the .gitignore, there's a limit to how many config files we can realistically have in root, so things that aren't really needed for the repo should probably go in .gitignore.

File: .nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+v15
```
Comment (MendyBerger) [inline .nvmrc:1]: .gitignore, same reason as .editorconfig

File: .env.sample [hunk 4]
```diff
@@ -149,15 +149,16 @@ DEV_AUTH=false
...
 # Optional overrides to force hitting remote server instead of local
 # These 3 should usually be set for frontend development
-LOCAL_API_URL="https://api.sandbox.jicloud.org"
-LOCAL_UPLOADS_URL="https://uploads.sandbox.jicloud.org"
-LOCAL_MEDIA_URL="https://media.jicloud.org"
-LOCAL_LEGACY_URL="https://legacy.jicloud.org"
+LOCAL_API_URL="https://api.sandbox.jigzi.org"
+LOCAL_UPLOADS_URL="https://uploads.sandbox.jigzi.org"
+LOCAL_MEDIA_URL="https://media.jigzi.org"
+LOCAL_LEGACY_URL="https://legacy.jigzi.org"
```
Comment (MendyBerger) [inline .env.sample:155]: `LOCAL_API_URL` change is great!

Others `LOCAL_UPLOADS_URL`, `LOCAL_MEDIA_URL`, and `LOCAL_LEGACY_URL` are using `jicloud` (we should probably change that but for now that's the case)

File: documentation/src/infrastructure/setup/requirements/frontend.md (hunk 1 above)
Comment (MendyBerger) [reply documentation/src/infrastructure/setup/requirements/frontend.md:16]: Than please remove it

**Review by MendyBerger — CHANGES_REQUESTED** [review]
Looks great overall! 

Added a few comments, feel free to push back

File: .env.sample (hunk 4 above)
Comment (johnnynotsolucky) [reply .env.sample]: Cool, that was a poor assumption on my part - I hadn't tested that those values worked yet for me.

**Review by MendyBerger — APPROVED** [review]

**Pull request merged**